corehq/apps/integration/payments/views.py

```diff
@@ -46,7 +82,30 @@ class PaymentsVerificationTableView(HqHtmxActionMixin, SelectablePaginatedTableV
     table_class = PaymentsVerifyTable
 
     def get_queryset(self):
-        return CaseSearchES().domain(self.request.domain).case_type(MOMO_PAYMENT_CASE_TYPE)
+        query = CaseSearchES().domain(self.request.domain).case_type(MOMO_PAYMENT_CASE_TYPE)
+        query = self._apply_filters(query)
+        return query
+
+    def _apply_filters(self, query):
+        if verification_status := self.request.GET.get('payment_verification_status'):
+            filter_value = 'True' if verification_status == PaymentVerificationStatusFilter.verified else ''
+            query = query.filter(case_property_query(PaymentProperties.PAYMENT_VERIFIED, filter_value))
+
+        if batch_number := self.request.GET.get('batch_number'):
+            query = query.filter(case_property_query(PaymentProperties.BATCH_NUMBER, batch_number))
+
+        if verified_by := self.request.GET.get('verified_by'):
+            query = query.filter(case_property_query(PaymentProperties.PAYMENT_VERIFIED_BY, verified_by))
+
+        if payment_status := self.request.GET.get('payment_status'):
+            if payment_status == PaymentStatus.submitted:
+                filter_value = 'True'
+            elif payment_status == PaymentStatus.submission_failed:
+                filter_value = 'False'
+            else:
+                filter_value = ''
```

Comment:
nit: Probably just a personal preference or overthinking, thought this would be more clear as
```
filter_value = None
if payment_status == PaymentStatus.submitted:
    filter_value = 'True'
elif payment_status == PaymentStatus.submission_failed:
    filter_value = 'False'
else payment_status == PaymentStatus.not_submitted:
    filter_value = ''
if filter_value:
    query = query.filter(case_property_query(PaymentProperties.PAYMENT_SUBMITTED, filter_value))
```